server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -467,6 +482,8 @@ private List<UploadedIndexMetadata> writeIndexMetadataParallel(ClusterState clus
             writeIndexMetadataAsync(clusterState, indexMetadata, latchedActionListener);
         }
 
+        invokeIndexMetadataUploadListeners(newIndexMetadataList, latch, exceptionList);
```

Comment:
Why are these called `listeners`?

Reply:
https://github.com/opensearch-project/OpenSearch/pull/13150#discussion_r1569018200. They are listening to index metadata upload event.